src/services/projects.service.ts

```diff
@@ -34,50 +34,60 @@ export class ProjectsService {
 	}
 
 	static async build(projectId: string, resources: Record<string, Uint8Array>): Promise<ServiceResponse<string>> {
-		const { error: resourcesError } = await this.setResources(projectId, resources);
-		if (resourcesError) {
-			LoggerService.error(`${namespaces.projectService} - Upload resources`, (resourcesError as Error).message);
+		try {
+			const { error: resourcesError } = await this.setResources(projectId, resources);
+			if (resourcesError) {
+				LoggerService.error(`${namespaces.projectService} - Upload resources`, (resourcesError as Error).message);
 
-			return { data: undefined, error: resourcesError };
-		}
+				return { data: undefined, error: resourcesError };
+			}
 
-		const { buildId, error } = await projectsClient.build({ projectId });
-		if (error) {
-			LoggerService.error(
-				`${namespaces.projectService} - Build: `,
-				convertErrorProtoToModel(error.value, projectId).message
-			);
+			const { buildId, error } = await projectsClient.build({ projectId });
+			if (error) {
+				LoggerService.error(
+					`${namespaces.projectService} - Build: `,
+					convertErrorProtoToModel(error.value, projectId).message
+				);
 
-			return { data: undefined, error };
+				return { data: undefined, error };
+			}
+			return { data: buildId, error: undefined };
+		} catch (error) {
+			LoggerService.error(namespaces.projectService, (error as Error).message);
+			return { data: undefined, error: (error as Error).message };
 		}
-		return { data: buildId, error: undefined };
 	}
 
 	static async deploy(projectId: string, buildId: string): Promise<ServiceResponse<string>> {
-		const { data: environments, error: envError } = await EnvironmentsService.listByProjectId(projectId);
-		if (envError) {
-			return { data: undefined, error: envError };
-		}
+		try {
```

Comment:
is this try catch has any meaning ?
you wrapped all your low level services with try catch and return {data, error} from them
why do you wrap it here again with try catch ?
what can happen ?

Reply:
If we sending invalid data on `projectsClient.build`, we can get an unhandled exception without any awareness from the user side. Something that happened when we sent invalid data on Windows.

Reply:
to remove